sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -175,6 +175,59 @@ object GpuCast extends Arm {
       cv.stringReplaceWithBackrefs(rule.search, rule.replace)
     })
   }
+
+  def sanitizeStringToIntegralType(input: ColumnVector, ansiEnabled: Boolean): ColumnVector = {
+    // Convert any strings containing whitespace to null values. The input is assumed to already
+    // have been stripped of leading and trailing whitespace
+    val sanitized = withResource(GpuScalar.from(null, DataTypes.StringType)) { nullVal =>
```

Comment:
We only need the null val for the I suspect somewhat uncommon case where there's at least one value with whitespace and we're not in ANSI mode.  Therefore we should only create this GPU-synchronizing resource in the case we need it rather than up-front for all cases.